### DISCUSSION TITLE
Recoil with JWT Authentication - Avoid re-render

### DISCUSSION THREAD

**Discussion by techknowfile** [facebookexperimental/Recoil#1853] [Q&A]
I'm using OAuth2/OIDC for AuthN/AuthZ, which returns a JWT accessToken with user details.

I use this token to initialize my User object, and it is also sent as an `Authorization` header in the requests to the backend.

The problem is that this token only lasts 15 minutes, so a refreshToken is used to update the accessToken.  With my current setup, User is a single atom, so every time the accessToken is refreshed, everything dependent on the user (the entire app) re-renders.

Is there a standard way to prevent re-rendering in situations such as this?  Will I need to separate the accessToken out from the user, and then programmatically update the User atom with an accessToken atom for only the changed values?

**Comment by drarmstr** [comment]
If you are using the token to initialize the User object which is an atom then you may want to use an Atom Effect for the initialization.  Atom Effects can initialize the state of an atom and reference other atoms without subscribing to them.
